### PR TITLE
refactor(downloader)!: Remove CVS support

### PR DESCRIPTION
**Pull request by sschuberth**
According to samples in ORT provenance databases, CVS usage is down to 1 of 350000. CVS binaries have not seen any updates since 2008, and its general use is rapidly declining. So remove CVS support from ORT to reduce maintenance effort and make the code base and dependencies to external tools a bit leaner.

> **Note**
> This is a breaking change for users of CVS, or dependencies whose source code is hosted in CVS. If required, the advice is to either scan source artifacts, create a curation for a Git mirror, or curate author / copyright plus license information and configure the scanner with `skipConcluded = true`.

### CODE REVIEW

File: utils/common/src/main/kotlin/Utils.kt [hunk 1]
```diff
@@ -31,9 +31,7 @@ val VCS_DIRECTORIES = listOf(
     ".git",
     ".hg",
     ".repo",
-    ".svn",
-    "CVS",
-    "CVSROOT"
+    ".svn"
```
Comment (fviernau) [inline utils/common/src/main/kotlin/Utils.kt:34]: Would it make sense to keep the entry here, as it is used as "exclude" patterns in several places, e.g. scanner? To me the entry seems to make sense no matter if the corresponding VCSes are supported.

**Review by fviernau — CHANGES_REQUESTED** [review]

File: utils/common/src/main/kotlin/Utils.kt (hunk 1 above)
Comment (sschuberth) [reply utils/common/src/main/kotlin/Utils.kt:34]: I was thinking about that, too, but when concluded that we should actually never come across these patterns if we're not able anymore to create CVS working trees.

Comment (fviernau) [reply utils/common/src/main/kotlin/Utils.kt:34]: Have you considered `ort scan --input-path` ?

Comment (sschuberth) [reply utils/common/src/main/kotlin/Utils.kt:34]: Actually, @mnonnenmacher recently proposed to drop that option, which is sometime I was planning to do soon. Are you actually using that option?

Comment (fviernau) [reply utils/common/src/main/kotlin/Utils.kt:34]: I've been using the option very rarely. I could as well use ScanCode directly instead in my use cases.

Comment (sschuberth) [reply utils/common/src/main/kotlin/Utils.kt:34]: Ok, so given that option would be dropped, are we good for this PR?

Comment (fviernau) [reply utils/common/src/main/kotlin/Utils.kt:34]: I've looked again a bit closer at all the usages of `VCS_DIRECTORIES`.
Namely,

1. `packZip()`
2. `calculatePackageVerificationCode()`
3. `ScanOss.scanPath()`
4. `Analyzer.findManagedFiles()`
5. `PackageManager.IGNORED_DIRECTORY_MATCHERS`

I still prefer these to work independently of whether a VCS is supported by ORT or not. 
So, let's just get another reviewers opinion to resolved this disagreement.

Comment (sschuberth) [reply utils/common/src/main/kotlin/Utils.kt:34]: > So, let's just get another reviewers opinion to resolved this disagreement.

There's not necessarily a disagreement. I probably just wasn't looking so deeply at the usages of `VCS_DIRECTORIES` as you did. Giving your findings, I'm actually fine with keeping CVS directories in there, even if it's still quite unlikely that we come across them.

**Review by fviernau — APPROVED** [review]

**Pull request merged**